runtime/src/chplmemtrack.c

```diff
@@ -239,7 +241,7 @@ static void addMemTableEntry(void* memAlloc, size_t number, size_t size, chpl_me
   memEntry->description = description;
   memEntry->memAlloc = memAlloc;
   memEntry->lineno = lineno;
-  memEntry->filename = filename; // do we want to copy this string?
+  memEntry->filename = chpl_lookupFilename(filename); // do we want to copy this string?
```

Comment:
Can we remove the trailing comment now?  It seems like it, since the strings come from a statically initialized table.
Orthogonally, would it make sense to store the index instead of the string, and look it up later when we print?  (I don't have a strong opinion on this and think it might not matter much.  Although, one could argue that recording into the memory table here should be as quick as possible, which would imply storing the index here, and looking up the string later.)